Feature/kas 4372 document card toon file created date en nooit piece created date updaten

StuyckRuben opened the pull request:
https://kanselarij.atlassian.net/browse/KAS-4372

Commit:
KAS-4372: piece.created replaced with piece.file.created + labels modified to show when a file has been modified

Commit:
KAS-4372: replaced piece.created with piece.file.created but kept piece.created as fallback

kobemertens commented on app/components/agenda/agenda-header/agenda-version-actions.hbs:
```diff
@@ -242,7 +242,7 @@
                   <div class="vlc-document-card-item__meta au-u-muted">
                     {{t "uploaded-at"}}
-                    {{datetime-at piece.created}}
+                    {{or (datetime-at piece.file.created) (datetime-at piece.created)}}
```
Why not `datetime-at (or piece.file.created piece.created)`?

kobemertens commented on app/components/documents/document-card.hbs:
```diff
@@ -99,7 +99,7 @@
                     </AuLinkExternal>
                     {{t "on"}}
-                    {{datetime-at this.piece.signedPieceCopy.created}}
+                    {{or (datetime-at this.piece.signedPieceCopy.file.created) (datetime-at this.piece.signedPieceCopy.created)}}
```
Same here

kobemertens commented on app/components/documents/document-card/edit-modal.hbs:
```diff
@@ -35,10 +35,10 @@
           <AuHelpText @skin="secondary">
             {{t "uploaded-at"}}
-            {{datetime-at @piece.created}}
+            {{or (datetime-at @piece.file.created) (datetime-at @piece.created)}}
```
Same

kobemertens commented on app/components/documents/document-preview/version-card.hbs:
```diff
@@ -17,7 +17,7 @@
         {{#if @piece.created}}
           {{t "uploaded-at"}}
-          {{datetime-at @piece.created "-"}}
+          {{or (datetime-at @piece.file.created) (datetime-at @piece.created) "-"}}
```
same

kobemertens commented on app/components/documents/linked-document-link.hbs:
```diff
@@ -25,10 +25,9 @@
+              {{#if this.lastPiece.file.created}}
                 {{t "uploaded-at"}}
-                {{datetime-at this.lastPiece.created}}
+                {{or (datetime-at this.lastPiece.file.created) (datetime-at this.lastpiece.created)}}
```
same

kobemertens commented on app/components/documents/linked-document-link.hbs:
```diff
@@ -94,10 +93,9 @@
+                    {{#if piece.file.created}}
                       {{t "uploaded-at"}}
-                      {{datetime-at piece.created}}
+                      {{or (datetime-at piece.file.created) (datetime-at piece.created)}}
```
same

kobemertens commented on app/components/documents/document-card/edit-modal.js:
```diff
@@ -172,5 +172,6 @@ export default class DocumentsDocumentCardEditModalComponent extends Component {
     this.isDeletingDerivedFile = false;
 
     this.args.onSave?.();
+    this.router.refresh('agenda.agendaitems.agendaitem.documents');
```
Refreshing a specific route from a component is a bad idea, we can't make any assumptions about where this component will be used. I would look into a `reload` of the relevant models, or refresh via the `onSave` function in the specific route where the component is used.

kobemertens reviewed (changes requested):
Suggested some minor changes

Commit:
resolve comments

sergiofenoll committed:
chore: remove unused router service

sergiofenoll committed:
chore: replace piece.reload with call to loadPieceRelatedData

Reloading the piece after editing does nothing for the document card,
because we show a queried version of the piece. Instead, after editing a
piece we should re-call the loadPieceRelatedData task to have fresh data.

sergiofenoll committed:
chore: use created-on as the default string in document card

In loadPieceRelatedData, created-on is the default anyway, so it's
better to use it as the default in the getter too so we don't have a
flash while it loads.

sergiofenoll committed:
chore: remove unused label argument

sergiofenoll committed:
chore: use same label for unedited pieces in history component

sergiofenoll committed:
chore: show modified in history if modified, else created date

sergiofenoll committed:
style: whitespace

sergiofenoll committed:
docs: document change in piece.save()

sergiofenoll committed:
Merge branch 'development' into feature/KAS-4372-document-card-toon-file-created-date-en-nooit-piece-created-date-updaten

ValenberghsSven commented on app/components/documents/document-preview/version-card.hbs:
```diff
@@ -17,7 +17,7 @@
         {{#if @piece.created}}
           {{t "uploaded-at"}}
-          {{datetime-at @piece.created "-"}}
+          {{datetime-at (or @piece.file.created @piece.created "-")}}
```
`"-"` outside of the or?
this is now an or with 3 params
The `datetime-at` helper accepts a datetime and an override for the "om" text to show date + time:
`helper(function datetimeAt([datetimeOrString, at = "om"])`
so `{{datetime-at (or @piece.file.created @piece.created) "-"}}`

sergiofenoll replied:
Hah, when I looked at this during review I assumed the `"-"` was there as a "fallback". If there's no `file.created` nor `piece.created`, show a dash. But this makes a lot more sense, and it's also in-line with the current behaviour. Will change.

ValenberghsSven commented on app/components/documents/linked-document-link.hbs:
```diff
@@ -25,10 +25,9 @@
+              {{#if this.lastPiece.file.created}}
                 {{t "uploaded-at"}}
-                {{datetime-at this.lastPiece.created}}
+                {{datetime-at (or this.lastPiece.file.created this.lastpiece.created)}}
```
if we have `this.lastPiece.file.created` we enter the if, so the second statement of the `datetime-at (or` will never be used

keep ` {{#if this.lastPiece.created}}` ? if there is a piece.created, we can check if there is a file.created and use that instead

ValenberghsSven commented on app/components/documents/document-card/edit-modal.hbs:
```diff
@@ -35,10 +35,10 @@
             </AuButton>
           </AuButtonGroup>
         </div>
-        {{#if @piece.created}}
```
keep the if as-is and just prefer piece.file.created if it exists?
Odd are good that if piece has a created that file will have one

ValenberghsSven commented on app/components/documents/document-version-history-item.js:
```diff
@@ -15,6 +17,20 @@ export default class DocumentsDocumentVersionHistoryItemComponent extends Compon
+    if (this.args.piece.created.getTime() == this.args.piece.modified.getTime()) {
+      return this.args.piece.created;
+    }
+    return this.args.piece.modified;
```
This uses the modified property instead of the `file.created`
The question arises: When to use `piece.modified` and when to use `piece.file.created`
We don't update `piece.modified` when replacing a `piece.file` in the backend services

sergiofenoll replied:
This is meant to follow the lines of the `document-card`, where we will always show `piece.modified`  if the piece has been editor (for regular pieces, for pieces with parts we use the part creation time).

I think those two are in line/doing the same thing?

ValenberghsSven reviewed (commented):
small changes and a question.

sergiofenoll committed:
chore: review comments

ValenberghsSven committed:
Merge branch 'development' of github.com:kanselarij-vlaanderen/frontend-kaleidos into feature/KAS-4372-document-card-toon-file-created-date-en-nooit-piece-created-date-updaten

ValenberghsSven committed:
KAS-4372 change document-card altDateToShow to be more dynamic, optional params where needed for legacy
